README.rst

```diff
@@ -12,77 +12,105 @@ smart_open — utils for streaming large files in Python
+``smart_open`` is a Python 2 & Python 3 library for **efficient streaming of very large files** from/to S3, HDFS, WebHDFS, HTTP, or local (compressed) files.
+It's a drop-in replacement for Python's built-in ``open()``: it can do anything ``open`` can (100% compatible, falls back to native ``open`` wherever possible), plus lots of nifty extra stuff on top.
 
 ``smart_open`` is well-tested, well-documented, and has a simple, Pythonic API:
 
 .. code-block:: python
 
-  >>> from smart_open import smart_open
-
+  >>> from smart_open import open
+  >>>
   >>> # stream lines from an S3 object
-  >>> for line in smart_open('s3://mybucket/mykey.txt', 'rb'):
-  ...    print(line.decode('utf8'))
+  >>> for line in open('s3://commoncrawl/robots.txt'):
+  ...    print(line, end='')
+  ...    break
+  User-Agent: *
 
+  >>> gz_path = 'smart_open/tests/test_data/crime-and-punishment.txt.gz'
+  >>>
   >>> # stream from/to compressed files, with transparent (de)compression:
-  >>> for line in smart_open('./foo.txt.gz', encoding='utf8'):
-  ...    print(line)
+  >>> for line in open(gz_path, encoding='utf-8'):
+  ...    print(line[:50])
+  ...    break
+  В начале июля, в чрезвычайно жаркое время, под веч
 
   >>> # can use context managers too:
-  >>> with smart_open('/home/radim/foo.txt.bz2', 'wb') as fout:
-  ...    fout.write(u"some content\n".encode('utf8'))
-
-  >>> with smart_open('s3://mybucket/mykey.txt', 'rb') as fin:
+  >>> bz2_path = gz_path.replace('.gz', '.bz2')
+  >>> with open(gz_path) as fin:
+  ...    with open(bz2_path, 'w') as fout:
+  ...        for line in fin:
+  ...           fout.write(line)
+
+  >>> # can use any IOBase operations, like seek
+  >>> with open('s3://commoncrawl/robots.txt', 'rb') as fin:
   ...     for line in fin:
-  ...         print(line.decode('utf8'))
+  ...         print(line.decode('utf-8'), end='')
+  ...         break
   ...     fin.seek(0)  # seek to the beginning
-  ...     b1000 = fin.read(1000)  # read 1000 bytes
-
-  >>> # stream from HDFS
-  >>> for line in smart_open('hdfs://user/hadoop/my_file.txt', encoding='utf8'):
-  ...     print(line)
+  ...     print(fin.read(4))
+  User-Agent: *
+  0
+  b'User'
 
   >>> # stream from HTTP
-  >>> for line in smart_open('http://example.com/index.html'):
-  ...     print(line)
+  >>> for line in open('http://example.com/index.html'):
+  ...     print(line, end='')
+  ...     break
+  <!doctype html>
 
-  >>> # stream from WebHDFS
-  >>> for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
-  ...     print(line)
+  >>> # stream from HDFS
+  >> for line in open('hdfs://user/hadoop/my_file.txt', encoding='utf8'):
```

Comment:
Is it normal to do `>>` (two symbols instead of three)? I never saw that before. It mismatches indent with other `>>>` lines (looks un-Pythonic).

Reply:
The two symbol thing comes from here: https://stackoverflow.com/questions/1809037/python-doctest-skip-entire-block

It's a hack that forces doctest to ignore that line.

Reply:
OK, thanks. Is the code block is copy-pastable though (ipython)?

Reply:
Unfortunately, no, it isn't pasteable.

If you'd like the examples to be pasteable as-is _and_ bypass tests, then we could provide code listings instead of REPL extracts. For example, instead of:

```python
>>> # read a file via WebHDFS
>>> for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
...     print(line)
```

we would show:

```python
# read a file via WebHDFS
for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
    print(line)
```

Is that an acceptable compromise?

Reply:
Yes, looks fine to me.